src/chainNode.ts

```diff
@@ -44,4 +62,57 @@ export default class ChainNode {
     const members = (await this.api.query.membership.members()).toJSON()
     return listParser.parse(members)
   }
+
+  async getAttachmentApiAddresses(account: string): Promise<OrgData[]> {
+    await this.api.isReady
+    // Call the organisationData RPC method with the appropriate parameters
+    const attachmentKey = 'AttachmentEndpoint'
+    const oidcKey = 'OidcConfigurationEndpoint'
+    const decodedLiterals: OrgData[] = []
+    const urlRegex = /^(https?:\/\/[^\s/$.?#].[^\s]*)$/i
+
+    const attachmentAddresses = await this.api.query.organisationData.orgData(account, attachmentKey)
+    const oidcAddresses = await this.api.query.organisationData.orgData(account, oidcKey)
+
+    if (attachmentAddresses && oidcAddresses) {
+      const jsonAddress = attachmentAddresses.toJSON()
+      const jsonOidcAddress = oidcAddresses.toJSON()
+
+      try {
+        const parsedOidcAddress = addressParser.parse(jsonOidcAddress)
+        const parsedAddress = addressParser.parse(jsonAddress)
+        const decodedAddressLiteral = hexToAscii(parsedAddress.literal)
+
+        const decodedOidcLiteral = hexToAscii(parsedOidcAddress.literal)
+
+        if (decodedAddressLiteral.match(urlRegex) && decodedOidcLiteral.match(urlRegex)) {
+          decodedLiterals.push({
+            account: account,
+            attachmentEndpointAddress: decodedAddressLiteral,
+            oidcConfigurationEndpointAddress: decodedOidcLiteral,
+          })
+        }
+      } catch (error) {
+        this.logger.error(`Error getting org data: ${error}`)
+      }
```

Comment:
I'd be tempted to refactor this bit out to a helper to reduce nesting. You can probably then parse each string separately which should make things cleaner